code-challenges.test.js

```diff
@@ -15,37 +15,146 @@
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = { upVotes: 13, downVotes: 2 }
-// Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
-// Expected output: -31
+// RED:
+
+describe("totalVotes", () => {
+  it("counts up and down votes return end tally", () => {
+    const votes1 = { upVotes: 13, downVotes: 2 }
+    // Expected output: 11
+    const votes2 = { upVotes: 2, downVotes: 33 }
+    // Expected output: -31
+    expect(totalVotes(votes1)).toEqual(11)
+    expect(totalVotes(votes2)).toEqual(-31)
+
+  })
+})
+// FAIL  ./code-challenges.test.js
+// totalVotes
+//   ✕ counts up and down votes return end tally
+
+// ● totalVotes › counts up and down votes return end tally
+
+//   ReferenceError: totalVotes is not defined
 
 // b) Create the function that makes the test pass.
 
+// PSUEDOCODE:
+// delcare a function called totalVotes
+// input: an object taht contains up votes and down votes
+// output: a number of the total up and down votes
+// Process: Inside of the function take the object and return the downvotes subtracted from upvotes.
+
+// GREEN:
+
+const totalVotes = (obj) => {
+  return obj.upVotes - obj.downVotes
+}
+
+console.log(totalVotes({ upVotes: 13, downVotes: 2 }))
+console.log(totalVotes({ upVotes: 2, downVotes: 33 }))
+// PASS  ./code-challenges.test.js
+// totalVotes
+//   ✓ counts up and down votes return end tally (1 ms)
+
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
+// RED:
+
+describe("bothDataArrays", () => {
+  it("takes in two arrays and returns one array", () => {
+    const dataArray1 = ["array", "object", "number", "string", "Boolean"]
+    const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
+    // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+    expect(bothDataArrays(dataArray1)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+    expect(bothDataArrays(dataArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+})
+// FAIL  ./code-challenges.test.js
+// bothDataArrays
+//   ✕ takes in two arrays and returns one array
+
+// ● bothDataArrays › takes in two arrays and returns one array
+
+//   ReferenceError: bothDataArrays is not defined
+
+// b) Create the function that makes the test pass.
+
+// GREEN:
+
+// PSUEDOCODE:
+// declare a function called bothDataArrays
+// input: an array of strings
+// output: a combined array with no duplicate values
+// Process: Reference used for code: https://stackoverflow.com/questions/1584370/how-to-merge-two-arrays-in-javascript-and-de-duplicate-items
```

Comment:
Okay but how do you understand this? Using this source, you were able to understand it better how?